robottelo/api/utils.py

```diff
@@ -130,6 +130,31 @@ def publish_puppet_module(puppet_modules, repo_url, organization_id=None):
     return cv.read()
 
 
+def import_puppet_module(sat):
+    """Download install and import puppet class. This is workaround for 6.10, there is no longer
+    content view
+
+    :param sat: satellite object
+    :return: puppet environment name,
+        environment name is likely to be searched in next steps of the test
+    """
+    env_name = 'test_environment'
```

Comment:
Keeping the constant name shouldn't fail this on running on the same satellite. Cant we just use `gen_string` from fauxfactory to have a randomly generated name every time ?

Reply:
no need for that, this is not failing, because command `puppet module install` will finish with module already installed
and on the other hand: let's say we installed the `api_class_parameters` first and then we want to install second module `cli_class_parameters`, that also works, cli is appended inside the test_environment, and api is still there.

Reply:
or better said puppet environment can contain multiple different puppet modules and same ones are not installed again, so it doesn't cause any issues

Reply:
@tstrych ACK !

Reply:
I just find out when it should be used. And why we should generate it to reduce test flakiness. 
Example.
You have one test with `api_class_parameters`
Second one with `cli_class_parameters`
Third one again with `api_class_parameters` and you forget to use the proper value when specifying the custom puppet module value, test will still pass because you already have them in `test_environment`. 
This will not be a big issue but we should test what we want to test, reduce the accidental test passes.
So thanks @jyejare . I will use generator.